### PR TITLE
Fix importing code in script mode

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
Fix importing code in script code by leveraging the script-mode assumption as opposed to the current working directory

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The main change is in `load_naive_entity` where we no longer add the working directory to `sys.path` prior to loading the python file, instead, we add the project root.

I also added a few opportunistic fixes, like removing commented code and ensuring that we check the existence of `__init__.py` files before we copy them.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2467

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/tools/script_mode.py [hunk 1]
```diff
@@ -142,7 +141,7 @@ def _find_project_root(source_path) -> Path:
     Traverse from current working directory until it can no longer find __init__.py files
     """
     # Start from the directory right above source_path
-    path = Path(source_path).parents[0]
+    path = Path(source_path).parents[0].resolve()
```
Comment (wild-endeavor) [inline flytekit/tools/script_mode.py:144]: what does the resolve() do?

**Review by wild-endeavor — DISMISSED** [review]
just a question though

File: flytekit/tools/script_mode.py (hunk 1 above)
Comment (eapolinario) [reply flytekit/tools/script_mode.py]: it turns a relative path into an absolute one. This particular one is useful to find the correct path of the script relative to the project root.

**Review by wild-endeavor — APPROVED** [review]
